configure.ac

```diff
@@ -612,6 +612,13 @@ if test "X$HDF_FORTRAN" = "Xyes"; then
   AC_PROG_FC([PAC_FC_SEARCH_LIST],)
   AC_F9X_MODS
 
+  ## Allow setting the fortran module install dir
+  AC_ARG_WITH([fmoddir],
```

Comment:
Need the CMake equivalent as well.

Reply:
So, I took a look at the CMake configuration, and I just don't think you have real FORTRAN support there in the first place.  You build h5fc from libhdf5cc.in which uses the stock libhdf5.pc file.  Is that going to contain the fortran libraries as well?

I'm just not very familiar with the HDF5 CMake build process to know for sure.

Reply:
@byrnHDF does not CMake already have Fortran_MODULE_DIRECTORY/CMAKE_Fortran_MODULE_DIRECTORY for this purpose.

Reply:
Yes, but we would need to allow an option to override the default. Plus I would need to override the option that was added last year to copy the fortran mod files to the include folder for install.

Reply:
So, I've added an initial HDF5_INSTALL_FMOD_DIR cmake change for discussion.   I have to say, from a Fedora packaging standpoint I find to whole /shared /static sub-directory thing very annoying.  We would prefer to build both static and shared libraries at the same time, but only install the shared executables as is done with the autotools build.  At least with gfortran the shared and static module files are identical.

Also, I think Fortran_MODULE_DIRECTORY/CMAKE_Fortran_MODULE_DIRECTORY is really only a build time variable.d

Reply:
Right, we added an option to copy the shared mod files to the include directory for the install. We can modify that to copy to the requested fortran folder. While gfortran on a linux machine might create the same file, other platforms and compilers do not; therefore the separate folders.